script.js

```diff
@@ -1,4 +1,709 @@
+  if (gameMode == GAME) {
+    // Assign number of dice first
+    // If input is valid (i.e. a digit >0), assign number of dice
+    if (numberOfDice == UNASSIGNED && checkNumber(input)) {
+      numberOfDice = Number(input);
+      console.log("numberOfDice", numberOfDice);
+    }
+    // If input is invalid, show error message
+    else if (numberOfDice == UNASSIGNED) {
+      return `Sorry, I do not understand. <br> <br> Please enter the number of dice you wish to play in this round.`;
+    }
+
+    // After number of dice has been assigned,
+    if (numberOfDice != UNASSIGNED) {
+      // Roll the amount of dice that was dictated earlier
+      diceRollArray = rollAnyQtyDice(numberOfDice);
+      console.log("diceRollArray", diceRollArray);
+      // Store the dice roll in an unsequenced array (for output message use)
+      var unsortedDiceRollArray = storeRandomArray(diceRollArray);
+      // Get best combined number of the dice rolled
+      var bestNumber = autoCombine(resultMode, diceRollArray);
+      // Store best combined number in the array along with other players' best numbers
+      competingNumbers.push(Number(bestNumber));
+      // Extract all of current player's best numbers thus far into the individual player's personal array of best numbers
+      var playerCount1 = playerNo;
+      console.log("playerNo", playerNo);
+      console.log("playerCount1", playerCount1);
+      var indivArray = [];
+      while (playerCount1 < competingNumbers.length) {
+        indivArray.push(competingNumbers[playerCount1]);
+        console.log("playerCount1", playerCount1);
+        console.log(
+          "competingNumbers[playerCount1]",
+          competingNumbers[playerCount1]
+        );
+        console.log("indivArray", indivArray);
+        playerCount1 += Number(numberOfPlayers);
+      }
+      console.log("competingNumbers", competingNumbers);
+      console.log("indivArray", indivArray);
+      // If there is none stored yet (i.e. still in first round), use push to get new data in
+      if (competingArrays.length < numberOfPlayers) {
+        competingArrays.push(indivArray);
+      }
+      // If there is already data stored within (i.e. second round and beyond), replace old data
+      if (competingArrays.length >= numberOfPlayers) {
+        competingArrays[playerNo] = indivArray;
+      }
+      console.log("competingArrays", competingArrays);
+
+      // Collate the total of all the numbers in the individual player's personal array
+      var indivTotal = addTotal(indivArray);
+      // If there is none stored yet (i.e. still in first round), use push to get new data in
+      if (competingTotals.length < numberOfPlayers) {
+        competingTotals.push(indivTotal);
+      }
+      // If there is already data stored within (i.e. second round and beyond), replace old data
+      if (competingTotals.length >= numberOfPlayers) {
+        competingTotals[playerNo] = indivTotal;
+      }
```

Comment:
Really interesting way to updating the competingTotals array
Good Job!! 

Great display and grit in tracking all your global variables